tools/i18n/extract.py

```
@@ -0,0 +1,369 @@
+                       help='Validate a locale file (e.g. zh_CN). Returns 0 if coverage=100%')
+    args = parser.parse_args()
+
+    # Validate mode
+    if args.validate:
+        result = validate_locale(args.validate)
+        if result is None:
+            sys.exit(1)
+        if result['coverage'] < 100:
+            print(f"\nFAILED: Coverage is {result['coverage']:.1f}%, expected 100%")
+            sys.exit(1)
+        print(f"\nPASSED: All strings translated")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)